common/djangoapps/third_party_auth/models.py

```diff
@@ -139,6 +279,23 @@ class ProviderConfig(ConfigurationModel):
             "authentication using the correct link is still possible."
         ),
     )
+    DATA_CONSENT_DISABLED = 'disabled'
+    DATA_CONSENT_OPTIONAL = 'optional'
+    DATA_CONSENT_REQUIRED = 'required'
+    DATA_CONSENT_STATE_CHOICES = (
+        (DATA_CONSENT_DISABLED, 'Disabled'),
+        (DATA_CONSENT_OPTIONAL, 'Optional'),
+        (DATA_CONSENT_REQUIRED, 'Required')
+    )
+    data_sharing_consent = models.CharField(
+        max_length=8,
+        blank=False,
+        choices=DATA_CONSENT_STATE_CHOICES,
+        default=DATA_CONSENT_DISABLED,
+        help_text=_(
+            "Whether this SSO requires data sharing permissions from users"
```

Comment:
Nit: No punctuation. ;)

Also, it might be worth making a note here that setting to "Disabled" will result in no data sharing with the provider (since it doesn't just disable the consent part, right?).